trimesh/visual/texture.py

```diff
@@ -19,10 +19,18 @@ def uv_to_color(uv, image):
     colors : (n, 4) float
       RGBA color at each of the UV coordinates
     """
-    x = (uv[:, 0] * image.width).round().astype(int)
-    y = ((1 - uv[:, 1]) * image.height).round().astype(int)
+    # https://shapenet.org/qaforum/index.php?qa=15&qa_1=why-is-the-texture-coordinate-in-the-obj-file-not-in-the-range  # NOQA
+    if not (uv.min() >= 0 and uv.max() <= 1):
+        uv = uv - np.floor(uv)
+
+    x = (uv[:, 0] * (image.width - 1)).round().astype(int)
+    y = ((1 - uv[:, 1]) * (image.height - 1)).round().astype(int)
```

Comment:
`-1` was missing in previous implementation. Now it supports uv value 0, 1. Previously, it causes out of index error at `image[y, x]`.